python/hail/matrixtable.py

```diff
@@ -1504,16 +1553,33 @@ def transmute_cols(self, **named_exprs) -> 'MatrixTable':
         Returns
         -------
         :class:`.MatrixTable`
-            Annotated matrix table.
         """
-        raise NotImplementedError()
+        named_exprs = {k: to_expr(v) for k, v in named_exprs.items()}
+        fields_referenced = set()
+        for k, v in named_exprs.items():
+            check_collisions(self._fields, k, self._col_indices)
```

Comment:
I think it might be good to add a function that returns the fields referenced and that takes an index as the parameter. That way you can reduce the amount of duplicated code between the 5 methods between MT and Table.

Reply:
will do